src/annotator/test/frame-observer-test.js

```diff
@@ -0,0 +1,138 @@
+    let frame = createAnnotatableIFrame();
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+
+    frame = createAnnotatableIFrame('enable-annotation', 'yes');
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+
+    frame = createAnnotatableIFrame('enable-annotation', 'true');
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+
+    frame = createAnnotatableIFrame('enable-annotation', '1');
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+
+    frame = createAnnotatableIFrame('enable-annotation', 'false'); // the actual value of the attribute is irrelevant
+    await waitForFrameObserver();
+
+    assert.calledWith(onFrameAdded, frame);
+  });
+
+  it("doesn't triggers onFrameAdded when non-annotatable iframes are added", async () => {
+    createAnnotatableIFrame('dummy-attribute');
+    await waitForFrameObserver();
+
+    assert.notCalled(onFrameAdded);
+  });
+
+  it('removal of the annotatable iframe triggers onFrameRemoved', done => {
+    sinon.stub(frameObserver, '_removeFrame').callThrough();
+    const frame = createAnnotatableIFrame();
+
+    waitForFrameObserver()
+      .then(() => {
+        assert.calledOnce(onFrameAdded);
+        assert.calledWith(onFrameAdded, frame);
+      })
+      .then(() => {
+        frame.remove();
+      });
+
+    waitForIFrameUnload(frame)
+      .then(() => waitForFrameObserver())
+      .then(() => {
+        assert.calledOnce(frameObserver._removeFrame);
+        assert.calledOnce(onFrameRemoved);
+        assert.calledWith(onFrameRemoved, frame);
+      })
+      .then(done);
+  });
+
+  it('removal of the `enable-annotation` attribute triggers onFrameRemoved', async () => {
+    const frame = createAnnotatableIFrame();
+    await waitForFrameObserver();
+
+    assert.calledOnce(onFrameAdded);
+    assert.calledWith(onFrameAdded, frame);
+
+    frame.removeAttribute('enable-annotation');
+    await waitForFrameObserver();
+
+    assert.calledOnce(onFrameRemoved);
+    assert.calledWith(onFrameRemoved, frame);
+  });
+
+  it('changing the `src` attribute triggers onFrameRemoved', async () => {
+    const frame = createAnnotatableIFrame();
+    await waitForFrameObserver();
+
+    assert.calledOnce(onFrameAdded);
+    assert.calledWith(onFrameAdded, frame);
+
+    frame.setAttribute('src', document.location);
+    await waitForIFrameUnload(frame);
+
+    assert.calledOnce(onFrameRemoved);
+    assert.calledWith(onFrameRemoved, frame);
+  });
+
+  // This test doesn't work. Surprisingly, `isAccessible` returns `true` even
+  // thought the iframe is from a different domain. My suspicion is that the
+  // iframe is accessed at a time where the loading has not yet started and it
+  // is therefore accessible.
+  // it("doesn't triggers onFrameAdded when annotatable iframe is from a different domain", async () => {
```

Comment:
This seems like something worth digging into to understand what is going on, in case there are ramifications we ought to be aware of.

Reply:
I did: https://github.com/hypothesis/client/pull/3686

Reply:
Since you know what causes the issue and there is an open PR, can you remove the commented-out code from this PR.